mod_cluster/extension/src/main/java/org/wildfly/extension/mod_cluster/DynamicLoadProviderResourceDefinition.java

```diff
@@ -22,49 +22,72 @@
 
 package org.wildfly.extension.mod_cluster;
 
+import static org.jboss.dmr.ModelType.EXPRESSION;
+
+import java.util.function.UnaryOperator;
+
 import org.jboss.as.clustering.controller.ChildResourceDefinition;
 import org.jboss.as.clustering.controller.ReloadRequiredResourceRegistration;
 import org.jboss.as.clustering.controller.ResourceDescriptor;
+import org.jboss.as.clustering.controller.SimpleAliasEntry;
 import org.jboss.as.controller.AttributeDefinition;
 import org.jboss.as.controller.ModelVersion;
+import org.jboss.as.controller.PathAddress;
 import org.jboss.as.controller.PathElement;
 import org.jboss.as.controller.SimpleAttributeDefinitionBuilder;
+import org.jboss.as.controller.operations.validation.IntRangeValidator;
 import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.as.controller.transform.TransformationContext;
+import org.jboss.as.controller.transform.description.AttributeConverter;
+import org.jboss.as.controller.transform.description.DiscardAttributeChecker;
+import org.jboss.as.controller.transform.description.RejectAttributeChecker;
 import org.jboss.as.controller.transform.description.ResourceTransformationDescriptionBuilder;
 import org.jboss.dmr.ModelNode;
 import org.jboss.dmr.ModelType;
 import org.jboss.modcluster.load.impl.DynamicLoadBalanceFactorProvider;
 
 /**
- * Definition for resource at address /subsystem=modcluster/proxy=X/dynamic-load-provider=configuration
+ * Definition for resource at address /subsystem=modcluster/proxy=X/load-provider=dynamic
+ * and its legacy alias /subsystem=modcluster/proxy=X/dynamic-load-provider=configuration
  *
  * @author Radoslav Husar
  */
 public class DynamicLoadProviderResourceDefinition extends ChildResourceDefinition<ManagementResourceRegistration> {
 
-    // TODO this should be a resource at path load-provider=dynamic...
-    public static final PathElement PATH = PathElement.pathElement("dynamic-load-provider", "configuration");
+    public static final PathElement LEGACY_PATH = PathElement.pathElement("dynamic-load-provider", "configuration");
+    public static final PathElement PATH = PathElement.pathElement("load-provider", "dynamic");
 
-    enum Attribute implements org.jboss.as.clustering.controller.Attribute {
-        DECAY("decay", ModelType.INT, new ModelNode(DynamicLoadBalanceFactorProvider.DEFAULT_DECAY_FACTOR)),
+    enum Attribute implements org.jboss.as.clustering.controller.Attribute, UnaryOperator<SimpleAttributeDefinitionBuilder> {
+        DECAY("decay", ModelType.DOUBLE, new ModelNode((double) DynamicLoadBalanceFactorProvider.DEFAULT_DECAY_FACTOR)),
         HISTORY("history", ModelType.INT, new ModelNode(DynamicLoadBalanceFactorProvider.DEFAULT_HISTORY)),
+        INITIAL_LOAD("initial-load", ModelType.INT, null) {
```

Comment:
Why no default value? e.g. 0.

Reply:
I didn't want to enable this by default initially, but lets do it.